### PR TITLE
Creating functionality to save data to JSON files

### PR DESCRIPTION
**Pull request by CarlosZ96**
Hello fellow code reviewer,

In this pull request, we have introduced a crucial feature that enhances our application's data management capabilities. We've implemented JSON file storage for object information, specifically for various entities like persons, books, and rentals.

What's New:

JSON File Storage: We've created a system where every time a new object of types 'person,' 'book,' or 'rental' is created within the application, its relevant information is automatically stored in a JSON file. This provides a structured and persistent means of retaining essential data.

Dynamic JSON File Creation: The JSON files will be generated dynamically when the application is run. This ensures that the data remains intact and readily accessible, even if the application is closed and reopened.

Why This Matters:

Improved Data Integrity: Storing data in a structured JSON format ensures data integrity and consistency, reducing the risk of information loss or corruption.

Persistence: With dynamic file creation, users can rely on the application to maintain their data across sessions. It enhances the user experience and trust in our product.

Scalability: As our application grows, this approach allows us to manage a wide range of data types efficiently.

### CODE REVIEW

**Review by zeff96 — COMMENTED** [review]
## STATUS: INVALID CODE REVIEW 🛑
Hi @CarlosZ96 and @jlberbesi , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

📢 Hey @CarlosZ96, it's crucial to highlight that for this milestone to be deemed valid, all partners should actively contribute to the codebase itself. While contributions to the README file are valuable, they aren't considered as direct contributions to the codebase. 📝🚀 So, please take note of this, @jlberbesi!

We kindly request you create a new branch from the 'develop' branch, ensuring equal and substantial contributions to the actual codebase. 🌱🙌 Commit your changes with clear and meaningful commit messages that reflect the purpose and impact of your work. Once done, please submit a Pull Request for review 📥. It's also essential to close the current Pull Request without merging to keep the process streamlined and aligned with our project goals. 🤝 Let's work together for a successful milestone! 🚀👥👍

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by CarlosZ96** [comment]
Hello @zeff96, I hope you have a good day, I appreciate the time you took to review our code, however, it is important to clarify that in addition to working together on a Zoom call, since I am the one presenting the screen, I suggested doing the code in my commit. This is by mutual agreement, so please do not reject the code, for this reason, we both participate in the creation of the code.

**Comment by zeff96** [comment]
Hello @CarlosZ96, I do understand your argument but it does not hold for project being considered as valid, kindly have a look at the project requirements again and be guided accordingly what is accepted as equal contribution. It would be nice for both of you to contribute to the actual codebase while on zoom call and each contribution should reflect on the commit messages

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @CarlosZ96, @jlberbesi, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.



## Invalid Code Review Request

📢 Hey @CarlosZ96, As mentioned by the previous reviewer it's crucial to highlight that for this milestone to be deemed valid, all partners should actively contribute to the codebase itself. While contributions to the README file are valuable, they aren't considered as direct contributions to the codebase. 📝🚀 So, please take note of this, @jlberbesi!

## 🔴🚫Attention🚫🔴 

📢 Your Code Review Request will be marked as invalid🚫 in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_**📢💡Invalid Code Review Request **does not count** into the code reviews limit.💡**_

File: main.rb
Comment (Efiamotu-1) [inline main.rb:1]: - [ ] When i try running your code to test your new changes, i am faced with the error below, please kindly look into your codebase and fix it as it important that i am able to run your code so i can test your recently added features :+1: 
rerp

File: README.md [hunk 1]
```diff
@@ -89,7 +89,7 @@ npm install
 - Twitter: [@twitterhandle](https://twitter.com/ELZambrano2)
 - LinkedIn: [@linkedinhandle](https://www.linkedin.com/in/carlos-zambrano-845406173/)
 
-👤 **Jose Luis Berbesi**
+👤 **Jose Luis Berbesi Lopera**
 
 - GitHub: [@githubhandle](https://github.com/jlberbesi)
 - Twitter: [@twitterhandle](https://twitter.com/imberbesi)
```
Comment (Efiamotu-1) [inline README.md:95]: - [ ] The linkedIn handle provided here is broken please kindly fix it, it is unprofessional to add broken links to your readme file :+1:

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @CarlosZ96 & @jlberbesi, 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jlberbesi** [comment]
@Efiamotu-1 Thanks a lot for your suggestions. I'm working now on it.

File: app.rb [hunk 2]
```diff
@@ -89,6 +88,49 @@ def list_person_rental
...
+    end
+  end
+
+  def load_rentals
+    return unless File.exist?('rentals.json')
+
+    JSON.parse(File.read('rentals.json')).each do |rental_data|
+      book = @books.find { |b| b.title == rental_data['book']['title'] }
+      person = @people.find { |p| p.id == rental_data['person']['id'] }
+      @rentals << Rental.new(rental_data['date'], book, person) if book && person
```
Comment (abdulhamiid) [inline app.rb]: On exit from your app, When I ran the app again, I encountered this error.

image

I checked the saved JSON for rentals and I observed that the issue is according to the name of the retrieved data which should be
`rental_data['person_id'] }` not `rental_data['person']['id'] }`
image

Please fix that and ensure that on startup of your app, it loads data from all files🛠🤗

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi Team 🙋‍♂️, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, I really appreciate the effort you put in! but there are still some which haven't been addressed yet.

almost-there

## Highlights 🌟

- ✔️ Run a program using the command line
- ✔️ Linters checks are passing
- ✔️ Store data in files.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — APPROVED** [review]
## Hi @CarlosZ96, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Implemented the features requested by previous code reviewer ✅ 
 
- Well-structured project ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**